crates/ruff_python_semantic/src/red_knot/semantic_index/definition.rs

```diff
@@ -0,0 +1,46 @@
+use crate::red_knot::semantic_index::ast_ids::{
+    LocalAnnotatedAssignmentId, LocalAssignmentId, LocalClassId, LocalFunctionId,
+    LocalImportFromId, LocalImportId, LocalNamedExprId,
+};
+
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub enum Definition {
+    Import(ImportDefinition),
+    ImportFrom(ImportFromDefinition),
+    ClassDef(LocalClassId),
+    FunctionDef(LocalFunctionId),
+    Assignment(LocalAssignmentId),
+    AnnotatedAssignment(LocalAnnotatedAssignmentId),
+    NamedExpr(LocalNamedExprId),
+    /// represents the implicit initial definition of every name as "unbound"
+    Unbound,
+    // TODO with statements, except handlers, function args...
+}
+
+impl From<LocalClassId> for Definition {
+    fn from(value: LocalClassId) -> Self {
+        Self::ClassDef(value)
+    }
+}
+
+impl From<LocalFunctionId> for Definition {
+    fn from(value: LocalFunctionId) -> Self {
+        Self::FunctionDef(value)
+    }
+}
```

Comment:
why no from impls for assignment, annassign, namedexpr?

Reply:
Too much work :P